util-images/network/netperfbenchmark/pkg/worker/util.go

```diff
@@ -0,0 +1,205 @@
+		if !strings.Contains(formattedString, zeroFormat+"-"+testDuration+fractionZeroFormat) {
+			continue
+		}
+		split := strings.Split(formattedString, " ")
+		// iperf gives aggregated result in a row with "SUM", but many fields are not getting aggregated
+		// hence not depending on iperf's aggregation
+		// if len(split) >= metricCount+1 && "SUM" != split[1] && split[2] == zeroFormat+"-"+testDuration+fractionZeroFormat {
+		if len(split) < metricCount+1 || "SUM" == split[1] || split[2] != zeroFormat+"-"+testDuration+fractionZeroFormat {
+			continue
+		}
+		// iperf sometimes prints duplicate rows for same session id(for same duration), making sure
+		// the row is taken for calculation only once.
+		if _, ok := sessionID[split[1]]; ok {
+			continue
+		}
+		for i, v := range split {
+			if i == 1 {
+				sessionID[v] = true
+				continue
+			}
+			//first index and hte last is ""
+			if i == 0 || i == 2 || i == metricCount+3 {
+				continue
+			}
+			tmp, err := strconv.ParseFloat(v, 64)
+			if err != nil {
+				klog.Errorf("Conversion error %v", err)
+				return nil, err
+			}
+			if len(aggregatedResult) < metricCount {
+				aggregatedResult = append(aggregatedResult, tmp)
+			} else {
+				switch operators[i] {
+				case "Sum":
+					aggregatedResult[i-3] = tmp + aggregatedResult[i-3]
+				case "Avg":
+					aggregatedResult[i-3] = (float64(count)*tmp + aggregatedResult[i-3]) / (float64(1 + count))
+				case "Min":
+					aggregatedResult[i-3] = math.Min(tmp, aggregatedResult[i-3])
+				case "Max":
+					aggregatedResult[i-3] = math.Max(tmp, aggregatedResult[i-3])
+				}
+			}
+			count++
+		}
+
+	}
+	klog.Infof("Final output: %v", aggregatedResult)
+	return aggregatedResult, nil
+}
+
+func parseSiegeResponse(result []string) ([]float64, error) {
+	result, err := trimSiegeResponse(result)
+	if err != nil {
+		return nil, err
+	}
+	aggregatedResult := make([]float64, 0, httpMetricsCount)
+	multiSpaceRegex := regexp.MustCompile(`\s+`)
+	for _, op := range result {
+		formattedString := multiSpaceRegex.ReplaceAllString(op, " ")
+		split := strings.Split(formattedString, ":")
+		klog.Infof("Formatted: %v", formattedString)
+		if len(split) > 1 {
+			split := strings.Split(split[1], " ")
+			if len(split) < 2 {
+				continue
+			}
+			tmp, err := strconv.ParseFloat(split[1], 64)
+			if err != nil {
+				klog.Errorf("Error parsing: %v", err)
+				return nil, err
+			}
+			aggregatedResult = append(aggregatedResult, tmp)
+		}
+	}
+	klog.Infof("Final output: %v", aggregatedResult)
+	return aggregatedResult, nil
+}
+
+func trimSiegeResponse(result []string) ([]string, error) {
+	var beginIndex, endIndex int
+	for beginIndex = 0; beginIndex < len(result) && !strings.HasPrefix(result[beginIndex], "Transactions:"); beginIndex++ {
+	}
+	if beginIndex == len(result) {
+		return nil, errors.New("Error parsing Siege response")
```

Comment:
unexpected Siege response: lack of `Transactions:` line

Reply:
Modified.